From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Siroshun09 <siro.shun8@gmail.com>
Date: Sun, 23 Jul 2023 06:13:19 +0900
Subject: [PATCH] Workaround for CME

java.util.ConcurrentModificationException: null
    at java.util.HashMap.computeIfAbsent(HashMap.java:1221) ~[?:?]
    at org.bukkit.craftbukkit.v1_20_R1.block.data.CraftBlockData.toBukkit(CraftBlockData.java:168) ~[okocraft-1.20.1.jar:git-Folia-"ff9deb2"]

diff --git a/src/main/java/org/bukkit/craftbukkit/block/data/CraftBlockData.java b/src/main/java/org/bukkit/craftbukkit/block/data/CraftBlockData.java
index c7b93b8d4ed15db0803e2304e2da70ed3624f13f..9833edc9725e3100aa3ebee08895be89335c9e8c 100644
--- a/src/main/java/org/bukkit/craftbukkit/block/data/CraftBlockData.java
+++ b/src/main/java/org/bukkit/craftbukkit/block/data/CraftBlockData.java
@@ -153,7 +153,7 @@ public class CraftBlockData implements BlockData {
         return exactMatch;
     }
 
-    private static final Map<Class<? extends Enum<?>>, Enum<?>[]> ENUM_VALUES = new HashMap<>();
+    private static final Map<Class<? extends Enum<?>>, Enum<?>[]> ENUM_VALUES = new java.util.concurrent.ConcurrentHashMap<>(); // okocraft - workaround for CME
 
     /**
      * Convert an NMS Enum (usually a BlockStateEnum) to its appropriate Bukkit
